src/PacksInitializable.sol

```diff
@@ -0,0 +1,861 @@
+    /// @dev Emits a CoSignerAdded event
+    function addCosigner(address cosigner_) external onlyRole(DEFAULT_ADMIN_ROLE) {
+        if (cosigner_ == address(0)) revert InvalidCosigner();
+        if (isCosigner[cosigner_]) revert AlreadyCosigner();
+        isCosigner[cosigner_] = true;
+        emit CosignerAdded(cosigner_);
+    }
+
+    /// @notice Removes an authorized cosigner
+    /// @param cosigner_ Address to remove as cosigner
+    /// @dev Only callable by admin role
+    /// @dev Emits a CoSignerRemoved event
+    function removeCosigner(address cosigner_) external onlyRole(DEFAULT_ADMIN_ROLE) {
+        if (!isCosigner[cosigner_]) revert InvalidCosigner();
+        isCosigner[cosigner_] = false;
+        emit CosignerRemoved(cosigner_);
+    }
+
+    /// @notice Sets the commit cancellable time.
+    /// @param commitCancellableTime_ New commit cancellable time
+    /// @dev Only callable by admin role
+    /// @dev Emits a CommitCancellableTimeUpdated event
+    function setCommitCancellableTime(uint256 commitCancellableTime_) external onlyRole(DEFAULT_ADMIN_ROLE) {
+        if (commitCancellableTime_ < MIN_COMMIT_CANCELLABLE_TIME) {
+            revert InvalidCommitCancellableTime();
+        }
+        uint256 oldCommitCancellableTime = commitCancellableTime;
+        commitCancellableTime = commitCancellableTime_;
+        emit CommitCancellableTimeUpdated(oldCommitCancellableTime, commitCancellableTime_);
+    }
+
+    /// @notice Sets the NFT fulfillment expiry time
+    /// @param nftFulfillmentExpiryTime_ New NFT fulfillment expiry time
+    /// @dev Only callable by admin role
+    /// @dev Emits a NftFulfillmentExpiryTimeUpdated event
+    function setNftFulfillmentExpiryTime(uint256 nftFulfillmentExpiryTime_) external onlyRole(DEFAULT_ADMIN_ROLE) {
+        if (nftFulfillmentExpiryTime_ < MIN_NFT_FULFILLMENT_EXPIRY_TIME) {
+            revert InvalidNftFulfillmentExpiryTime();
+        }
+        uint256 oldNftFulfillmentExpiryTime = nftFulfillmentExpiryTime;
+        nftFulfillmentExpiryTime = nftFulfillmentExpiryTime_;
+        emit NftFulfillmentExpiryTimeUpdated(oldNftFulfillmentExpiryTime, nftFulfillmentExpiryTime_);
+    }
+
+    /// @notice Sets the maximum allowed reward
+    /// @param maxReward_ New maximum reward value
+    /// @dev Only callable by admin role
+    function setMaxReward(uint256 maxReward_) external onlyRole(OPS_ROLE) {
+        if (maxReward_ < minReward) revert InvalidReward();
+
+        uint256 oldMaxReward = maxReward;
+        maxReward = maxReward_;
+        emit MaxRewardUpdated(oldMaxReward, maxReward_);
+    }
+
+    /// @notice Sets the minimum allowed reward
+    /// @param minReward_ New minimum reward value
+    /// @dev Only callable by admin role
+    function setMinReward(uint256 minReward_) external onlyRole(DEFAULT_ADMIN_ROLE) {
+        if (minReward_ > maxReward) revert InvalidReward();
+        if (minReward_ < minReward) revert InvalidReward();
```

Comment:
I think this means the new min reward cannot be lower than the existing min reward.

Reply:
Good catch, this was a bug.